datafusion/spark/src/function/math/modulus.rs

```diff
@@ -87,9 +85,11 @@ impl ScalarUDFImpl for SparkMod {
     }
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        if arg_types.len() != 2 {
-            return internal_err!("mod expects exactly two arguments");
-        }
+        assert_eq_or_internal_err!(
+            arg_types.len(),
+            2,
+            "mod expects exactly two arguments"
+        );
```

Comment:
```suggestion
```

We shouldn't be checking this in return type since signature guards this for us already